mrbgems/mruby-io/src/file.c

```diff
@@ -283,9 +283,11 @@ static int
 mrb_file_is_absolute_path(const char *path)
 {
 #ifdef _WIN32
-#define IS_PATHSEP(x) (x == '/' || x == '\')
-  if (strlen(path) < 3) return 0;
-  return (isalpha(path[0]) && path[1] == ':' && IS_PATHSEP(path[2]));
+#define IS_PATHSEP(x) (x == '/' || x == '\\')
+  if (isalpha(path[0]))
+    return (strlen(path) > 2 && path[1] == ':' && IS_PATHSEP(path[2]));
+  else
+    return (IS_PATHSEP(path[0]) && IS_PATHSEP(path[1]));
```

Comment:
It seems that we need to add `path[0] != '\0'` check.

Reply:
I think that there is no problem as it is...

Reply:
watching over the length of the string might be more appropriate. when trying to retrieve `path[1]` it will collapse.

Reply:
Pull Request #4952

Reply:
Ah, sorry. I was wrong. `path[0] != '\0'` is needless.